soc/nordic/ironside/counter.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```

Comment:
```suggestion

#include <stddef.h>
```
Fails with the following error:
```
zephyr/soc/nordic/ironside/counter.c:36:22: error: 'NULL' undeclared (first use in this function)
   36 |         if (value == NULL) {
```